### PR TITLE
unwrap updates

### PR DESCRIPTION
**Pull request by sethhrbek**

### CODE REVIEW

File: service/src/utils/onchain-data.ts [hunk 1]
```diff
@@ -122,7 +131,7 @@ export async function unwrapSarcophagus(web3Interface: Web3Interface, sarcoId: s
   try {
     await web3Interface.archaeologistFacet.unwrapSarcophagus(
       sarcoId,
-      ethers.utils.arrayify(decryptedShard) // TODO: Might need an extra `Buffer.from`. `unwrapSarcophagus` expects a solidity bytes type, `decryptedShard` is a hex string
+      decryptedShard
```
Comment (DarksightKellar) [inline service/src/utils/onchain-data.ts:134]: Interesting. The '0x`- prefixed hex string is treated as `bytes` in the contract?

File: service/src/utils/onchain-data.ts [hunk 2]
```diff
@@ -60,21 +65,26 @@ export async function getOnchainCursedSarcophagi(
...
     const sarco = await web3Interface.viewStateFacet.getSarcophagus(sarcoId);
     const archStorage = await web3Interface.viewStateFacet.getSarcophagusArchaeologist(
       sarcoId,
       web3Interface.ethWallet.address
     );
 
-    if (sarco.state === SarcophagusState.Exists && !archStorage.unencryptedShard) {
+    // If sarcophagus exists and the unencrypted shard is empty, then it (potentially)
+    // still needs to be unwrapped
+    if ((sarco.state === SarcophagusState.Exists) && archStorage.unencryptedShard === '0x') {
```
Comment (DarksightKellar) [inline service/src/utils/onchain-data.ts:83]: Solidity defaults! Hah.

JS' "falsy" logic is such a recipe for disaster.

As an aside: Are those extra parentheses needed around `sarco.state === SarcophagusState.Exists`

Comment (sethhrbek) [reply service/src/utils/onchain-data.ts]: @DarksightKellar thanks those parens arent needed --- removed them

File: service/src/utils/onchain-data.ts (hunk 1 above)
Comment (DarksightKellar) [reply service/src/utils/onchain-data.ts:134]: Very good to know

**Pull request merged**